### ISSUE TITLE
No support for document fragments

### ISSUE THREAD

**Issue by GrantGryczan** [gilbarbara/react-from-dom#13]
## Issue Description

Passing a `DocumentFragment` (which is a `Node` with `nodeType === 11`) into the `convert` function doesn't parse anything inside it. Instead, the entire input is discarded as `null`. Thus, I believe there is currently no built-in way to parse a number of `Node`s without a parent.

Ideally, `DocumentFragment`s should be converted into arrays (or `React.Fragment`s if that's more practical for whatever reason).

Here is some information from MDN about document fragments: https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment

(It may be worth noting that it's impossible for document fragments to be a child node. They can only be a root node, the value passed directly into the `convertFromNode` function. I say this because you may be able to optimize using that information.)

## Example Input

```tsx
const fragment = document.createDocumentFragment();
fragment.appendChild(document.createTextNode('Hello, world!'));
const div = document.createElement('div');
div.appendChild(document.createTextNode('I\'m inside a `div`.'));
fragment.appendChild(div);

convert(fragment);
```

## Expected Output (Approximate)

```tsx
[
    'Hello, world!',
    <div>
        I'm inside a `div`.
    </div>
]
```

## Actual Output

```tsx
null
```

## Temporary Workaround

I have implemented a temporary workaround as an action:

```tsx
{
    // This action is to support document fragments. It can be removed when https://github.com/gilbarbara/react-from-dom/issues/13 is fixed.
    condition: node => node.nodeType === 11,
    post: node => (
        [...node.childNodes].map((childNode, i) => (
            <Fragment key={i}>
                {convertFromNode(childNode, convertFromNodeOptions)}
            </Fragment>
        ))
    )
}
```

Though I'm sure a more optimized solution could arise from built-in support.

## My Personal Use Case

I am using [DOMPurify](https://github.com/cure53/DOMPurify)'s `RETURN_DOM_FRAGMENT` option which returns a `DocumentFragment` after sanitizing the inputted HTML string. I would like to pass this sanitized node into `react-from-dom`'s parser directly without appending it into another element.

**Comment by GrantGryczan** [comment]
I have implemented a temporary workaround as an action:
```tsx
{
    // This action is to support document fragments. It can be removed when https://github.com/gilbarbara/react-from-dom/issues/13 is fixed.
    condition: node => node.nodeType === 11,
    post: node => (
        [...node.childNodes].map((childNode, i) => (
            <Fragment key={i}>
                {convertFromNode(childNode, convertFromNodeOptions)}
            </Fragment>
        ))
    )
}
```

**Comment by gilbarbara** [comment]
hey @GrantGryczan 
The actions exist for this exact situation...

But if you want to create a PR that wouldn't add a lot of complexity/code, that's ok.

**Comment by gilbarbara** [comment]
Fixed in 0.7.0
